src/main/java/edu/cornell/kfs/concur/service/impl/ConcurEventNotificationServiceImpl.java

```diff
@@ -65,30 +65,40 @@ protected String truncateValidationResultMessageToMaximumDatabaseFieldSize(Strin
-                    try {
-                        ConcurEventNotification concurEventNotification = getConcurEventNotificationConversionService().convertConcurEventNotification(concurEventNotificationDTO);
-                        saveConcurEventNotification(concurEventNotification);
-                        getConcurReportsService().deleteFailedEventQueueItemInConcur(findNotificationId(concurEventNotification.getNotificationURI()));
-                    } catch (ParseException e) {
-                        LOG.error("validate():" + e.getMessage(), e);
+            if (controller.logFailedEventQueue) {
+                String notificationListString = getConcurReportsService().retrieveFailedEventQueueNotificationsFromConcurAsString();
+                LOG.info("retrieveAndPersistFailedEventQueueReports, notificationListString: " + notificationListString);
+            }
+            if (controller.writeFailedEventQueue) {
+                if (areThereNotificationsToProcess(notificationList)) {
+                    LOG.info("retrieveAndPersistFailedEventQueueReports, found " + notificationList.getConcurEventNotificationDTOs().size() + " failed events to process.");
+                    for (ConcurEventNotificationDTO concurEventNotificationDTO : notificationList.getConcurEventNotificationDTOs()) {
+                        try {
+                            ConcurEventNotification concurEventNotification = getConcurEventNotificationConversionService().convertConcurEventNotification(concurEventNotificationDTO);
+                            saveConcurEventNotification(concurEventNotification);
+                            getConcurReportsService().deleteFailedEventQueueItemInConcur(findNotificationId(concurEventNotification.getNotificationURI()));
+                        } catch (ParseException e) {
+                            LOG.error("retrieveAndPersistFailedEventQueueReports, error trying to save items: " + e.getMessage(), e);
+                        }
                     }
+                } else {
+                    LOG.info("retrieveAndPersistFailedEventQueueReports, There were no failed events to process.");
                 }
             } else {
-                LOG.info("retrieveAndPersistFailedEventQueueReports, There were no failed events to process.");
+                LOG.info("retrieveAndPersistFailedEventQueueReports, persisting failed event queue items is turned off.");
             }
         } else {
-            LOG.info("retrieveAndPersistFailedEventQueueReports, failed event queue processing is turned off");
+            LOG.info("retrieveAndPersistFailedEventQueueReports, failed event queue processing is turned off.");
         }
     }
     
-    private boolean shouldProcessFailedEventQueue() {
+    protected ConcurFailedEventQueueProcessingController findCOncurFailedEventQueueProcessingController() {
```

Comment:
Minor capitalization typo in method name; should be changed from "...findCOncur..." to "...findConcur..."

Reply:
OK, will fix